src/mysql_conn.erl

```diff
@@ -314,9 +314,19 @@ handle_call({change_user, Username, Password, Options}, From,
     end;
 handle_call(reset_connection, _From, #state{socket = Socket, sockmod = SockMod} = State) ->
     setopts(SockMod, Socket, [{active, false}]),
-    Ok = mysql_protocol:reset_connnection(SockMod, Socket),
+    Result = mysql_protocol:reset_connnection(SockMod, Socket),
     setopts(SockMod, Socket, [{active, once}]),
-    {reply, ok, update_state(Ok, State)};
+    State1 = update_state(Result, State),
+    Reply = case Result of
+        #ok{} -> ok;
+        #error{} = E ->
+            %% 'COM_RESET_CONNECTION' is added in MySQL 5.7,
+            %% and return "Unkown command" in 5.6 (earlier version included).
+            State1#state.log_warnings andalso log_warnings(State1, "RESET CONNECTION"),
```

Comment:
If there is an error, there are no warnings, but for an OK message there can be warnings from the server.

In other places, this logging is just after update_state, e.g. for CHANGE USER:

```Erlang
    State1 = update_state(Result, State),
    State1#state.warning_count > 0 andalso State1#state.log_warnings
        andalso log_warnings(State1, "CHANGE USER"),
```

If you look at `update_state/1`, where `warning_count` is updated from the result. If the packet is an error packet, the warning count is set to zero because the error packet doesn't contain any warning count.

Reply:
Warn (or error) message is helpful. 
Developers want to know if it is effective and what happened when the function reset_connection is called.